code/modules/mining/machine_local_teleporter.dm

```diff
@@ -0,0 +1,41 @@
+/obj/machinery/mineral/local_teleporter
+	name = "local teleporter"
+	desc = "A floor-mounted local area teleporter. Due to the small configuration, it can only teleport anything that crosses a short distance."
+	icon = 'icons/obj/mining.dmi'
+	icon_state = "local_teleporter"
+	density = FALSE
+	anchored = TRUE
+	use_power = FALSE
+	active_power_usage = 500
+	var/obj/machinery/mineral/local_teleporter/linked_teleporter
+	var/connected_overlay
+
+/obj/machinery/mineral/local_teleporter/Initialize(mapload)
+	..()
+	connected_overlay = icon(icon, "[initial(icon_state)]-connected")
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/machinery/mineral/local_teleporter/LateInitialize()
+	var/turf/T = get_step(src, dir)
+	linked_teleporter = locate() in get_step(T, dir)
```

Comment:
You could make this machine more versatile for other mappers in future by giving it a var/teleport_range and something like this:

```suggestion
	linked_teleporter = locate() in orange(teleport_range, src)
```

Reply:
currently this is configured to only work in a straight line, so having it be orange wouldn't really work or be desirable